packages/components/organisms/f-checkout/src/components/Selector.vue

```diff
@@ -40,36 +40,50 @@ export default {
         * Returns an array of formatted dates to display as options in a dropdown
         */
         fulfilmentTimes () {
-            const times = this.availableFulfilment.isAsapAvailable
-                ? [this.$t('asapFulfilmentOption')]
-                : [];
-
+            const times = [];
             this.availableFulfilment.times.forEach(time => {
                 const formattedTime = this.$d(new Date(time.from), 'short');
 
-                times.push(formattedTime);
+                times.push({
+                    text: formattedTime,
+                    value: time.from
+                });
             });
 
+            if (this.availableFulfilment.isAsapAvailable && times.length) {
+                times[0].text = this.$t('asapFulfilmentOption');
+            }
+
             return times;
         }
     },
 
     watch: {
-        'fulfilmentTimes' () {
-            if (this.fulfilmentTimes.length > 0) {
-                this.selectionChanged(this.fulfilmentTimes[0]);
+        fulfilmentTimes (newFulfilmentTimes) {
+            if (newFulfilmentTimes.length) {
```

Comment:
Do you need the length check now? The watcher should only trigger when there's a change to the times now so i assume it will always be populated unless it ever gets cleared to a empty array?

Reply:
That's a good point. I can't think of a situation where we try to push an empty array but if that does happen in future we can fix it then. It's arguably more predictable without the length check because otherwise the value that gets submitted might be out-of-sync with what's in the dropdown.